03-dplyr.Rmd

```diff
@@ -548,140 +541,133 @@ surveys_gw <- surveys %>%
 >    each measurement per year in different plot types. Let's walk through a
->    common solution for this type of problem. First, use `gather()` to create a
+>    common solution for this type of problem. First, use `pivot_longer()` to create a
 >     dataset where we have a key column called `measurement` and a
 >    `value` column that takes on the value of either `hindfoot_length` or
 >    `weight`. *Hint*: You'll need to specify which columns are being gathered.
 >
 > ```{r, answer=TRUE, purl=FALSE}
-> surveys_long <- surveys %>%
->   gather("measurement", "value", hindfoot_length, weight)
+>surveys_long <- surveys %>%
+> pivot_longer(names_to = "measurement", values_to = "value", cols = c(hindfoot_length, weight))
+>
 > ```
 >
 > 4. With this new data set, calculate the average of each
 >    `measurement` in each `year` for each different `plot_type`. Then
->    `spread()` them into a data set with a column for `hindfoot_length` and
+>    `pivot_wider()` them into a data set with a column for `hindfoot_length` and
 >    `weight`. *Hint*: You only need to specify the key and value
->    columns for `spread()`.
+>    columns for `pivot_wider()`.
 >
 > ```{r, answer=TRUE, purl=FALSE}
 > surveys_long %>%
 >   group_by(year, measurement, plot_type) %>%
 >   summarize(mean_value = mean(value, na.rm=TRUE)) %>%
->   spread(measurement, mean_value)
+>   pivot_wider(names_from = measurement, values_from = mean_value)
 > ```
 
+
+
 ```{r, eval=FALSE, purl=TRUE, echo=FALSE}
 ## Reshaping challenges
 
-## 1. Spread the `surveys` data frame with `year` as columns, `plot_id` as rows, and the number of genera per plot as the values. You will need to summarize before reshaping, and use the function `n_distinct()` to get the number of unique genera within a particular chunk of data. It's a powerful function! See `?n_distinct` for more.
+## 1. Reshape the `surveys` data frame with `year` as columns, `plot_id` as rows, and the number of genera per plot as the values. You will need to summarize before reshaping, and use the function `n_distinct()` to get the number of unique genera within a particular chunk of data. It's a powerful function! See `?n_distinct` for more.
+
+
+## 2. Now take that data frame and `pivot_longer()` it again, so each row is a unique `plot_id` by `year` combination.
 
-## 2. Now take that data frame and `gather()` it again, so each row is a unique `plot_id` by `year` combination.
+## 3. The `surveys` data set has two measurement columns: `hindfoot_length` and `weight`. This makes it difficult to do things like look at the relationship between mean values of each measurement per year in different plot types. Let's walk through a common solution for this type of problem. First, use `pivot_longer()` to create a dataset where we have a key column called `measurement` and a `value` column that takes on the value of either `hindfoot_length` or `weight`. *Hint*: You'll need to specify which columns are being gathered.
```

Comment:
```suggestion
## 3. The `surveys` data set has two measurement columns: `hindfoot_length` and `weight`. This makes it difficult to do things like look at the relationship between mean values of each measurement per year in different plot types. Let's walk through a common solution for this type of problem. First, use `pivot_longer()` to create a dataset where we have a names column called `measurement` and a values column that takes on the value of either `hindfoot_length` or `weight`. *Hint*: You'll need to specify which columns will be part of the reshape.
```

Reply:
my previous comment applies here, as the lesson and code handout  should be kept in synchrony.